cmd/placement-translator/main.go

```diff
@@ -169,29 +176,32 @@ func main() {
 	kcpClusterClientset, err := kcpclusterclientset.NewForConfig(baseRestConfig)
 	if err != nil {
 		logger.Error(err, "Failed to create all-cluster clientset for kcp APIs")
-		os.Exit(10)
+		os.Exit(60)
 	}
 
 	discoveryClusterClient, err := clusterdiscovery.NewForConfig(baseRestConfig)
 	if err != nil {
 		logger.Error(err, "Failed to create all-cluster discovery client")
-		os.Exit(20)
+		os.Exit(70)
 	}
 
 	dynamicClusterClient, err := clusterdynamic.NewForConfig(baseRestConfig)
 	if err != nil {
 		logger.Error(err, "Failed to create all-cluster dynamic client")
-		os.Exit(30)
+		os.Exit(80)
 	}
 
 	dynamicClusterInformerFactory := clusterdynamicinformer.NewDynamicSharedInformerFactory(dynamicClusterClient, 0)
 
-	crdClusterInformer := dynamicClusterInformerFactory.ForResource(apiextensionsv1.SchemeGroupVersion.WithResource("customresourcedefinitions"))
-	bindingClusterInformer := dynamicClusterInformerFactory.ForResource(apisv1alpha1.SchemeGroupVersion.WithResource("apibindings"))
+	crdClusterPreInformer := dynamicClusterInformerFactory.ForResource(apiextensionsv1.SchemeGroupVersion.WithResource("customresourcedefinitions"))
+	// crdClusterPreInformer.Informer().Cluster()
+	bindingClusterPreInformer := dynamicClusterInformerFactory.ForResource(apisv1alpha1.SchemeGroupVersion.WithResource("apibindings"))
 
 	doneCh := ctx.Done()
 	// TODO: more
-	pt := placement.NewPlacementTranslator(concurrency, ctx, epClusterPreInformer, spsClusterPreInformer, mbwsPreInformer, kcpClusterClientset, discoveryClusterClient, crdClusterInformer, bindingClusterInformer, dynamicClusterClient)
+	pt := placement.NewPlacementTranslator(concurrency, ctx, epClusterPreInformer, spsClusterPreInformer, syncfgClusterPreInformer,
+		mbwsPreInformer, kcpClusterClientset, discoveryClusterClient, crdClusterPreInformer, bindingClusterPreInformer,
+		dynamicClusterClient, edgeClusterClientset)
 	edgeInformerFactory.Start(doneCh)
 	espwInformerFactory.Start(doneCh)
 	dynamicClusterInformerFactory.Start(doneCh)
```

Comment:
NewPlacementTranslator() waits returning until Informer.HasSynced is true. I might not correctly understand the behavior of informer but can the informer be a synced status (;Informer.HasSynced is true) before the informer factory starts?

Reply:
Yes, this change was a mistake.  I have verified that in testing, and am about to push a fix.

Reply:
Fix pushed.